### DISCUSSION TITLE
Error "sending query failed: another command is already in progress" at AsyncConnectionPool

### DISCUSSION THREAD

**Discussion by lexorama** [psycopg/psycopg#226] [Q&A]
Hi !
I use
 AsyncConnectionPool:
```python
pool = AsyncConnectionPool(conninfo=conninfo_,
                                                                       min_size=20,
                                                                       max_size=30,
                                                                       reconnect_timeout=2
                                                                       )
```

query:
``` python
query = sql.SQL('INSERT INTO all_streams(value_jsonb,originaltime,guid,stream) VALUES ({},to_timestamp({}),{},{})').
         format(request.valueJSONB,request.originalTime,request.GUID,request.stream)
```

execute:
```python
            try:
                async with pool.connection(timeout=0.03) as conn:
                    async with conn.transaction():
                        await conn.execute(query)
                return
             except psycopg.OperationalError as ex:
                print(f'SQL_Exec: OperationalError {ex.sqlstate}/{ex.args[0]}')
```

There is a constant stream of 10TPS transactions to the service. And from time to time, 3-4 times a day I get this error:
>error ignored rolling back transaction on <psycopg.AsyncConnection [ACTIVE] (host=192.168.75.210 user=rex database=postgres) at 0x7f017c521550>: sending query failed: another command is already in progress
closing returned connection: <psycopg.AsyncConnection [ACTIVE] (host=192.168.75.210 user=rex database=postgres) at 0x7f017c521550>
discarding closed connection: <psycopg.AsyncConnection [BAD] at 0x7f017c521550>

?

Can someone explain me what I'm doing wrong?

**Comment by dvarrazzo** [comment]
Why are you formatting the query instead of passing the parameters to execute?

Apart from that I don't see anything obviously wrong. However there was a bug closed in 3.0.5 regarding rollbacks in transaction block. What version are you using?

**Reply by lexorama** [reply]
> Why are you formatting the query instead of passing the parameters to execute?

Queries are very different, the formatting option is the most universal.

> 
> Apart from that I don't see anything obviously wrong. However there was a bug closed in 3.0.5 regarding rollbacks in transaction block. What version are you using?

requirements.txt:
psycopg-binary==3.0.1
psycopg-pool==3.0
psycopg==3.0.1

I'll try 3.0.5.

Thank you so much!

WBR, Alex
